src/osd/OSD.cc

```diff
@@ -9191,17 +9193,18 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
   sdata->sdata_op_ordering_lock.Lock();
 
   auto q = sdata->pg_slots.find(item.first);
-  if (q == sdata->pg_slots.end() ||
-      q->second.to_process.empty()) {
-    dout(20) << __func__ << " " << item.first
-	     << " no pg_slot or nothing pending" << dendl;
+  assert(q != sdata->pg_slots.end());
+  auto& slot = q->second;
+  --slot.num_running;
```

Comment:
This and the increment are placed bizarrely. Why not just do it right next to the lock twiddling, preferably within a scope that makes it really obvoius?